flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/source/FileSplitMonitoringFunctionITCase.java

```diff
@@ -0,0 +1,279 @@
+			hdfsCluster = builder.build();
+
+			hdfsURI = "hdfs://" + hdfsCluster.getURI().getHost() + ":" + hdfsCluster.getNameNodePort() +"/";
+			hdfs = new org.apache.hadoop.fs.Path(hdfsURI).getFileSystem(hdConf);
+
+		} catch(Throwable e) {
+			e.printStackTrace();
+			Assert.fail("Test failed " + e.getMessage());
+		}
+	}
+
+	@After
+	public void destroyHDFS() {
+		FileUtil.fullyDelete(baseDir);
+		hdfsCluster.shutdown();
+	}
+
+	//						END OF PREPARATIONS
+
+	private static final Object lock = new Object();
+
+	private boolean[] finished;
+
+	@Override
+	protected void testProgram() throws Exception {
+		FileCreator fileCreator = new FileCreator(INTERVAL);
+		Thread t = new Thread(fileCreator);
+		t.start();
+		Thread.sleep(100);
+
+		StringFileFormat format = new StringFileFormat();
+		format.setFilePath(hdfsURI);
+
+		Configuration config = new Configuration();
+		config.setString("input.file.path", hdfsURI);
+
+		FileSplitMonitoringFunction<String> monitoringFunction = null;
+		FileSplitReadOperator<String> reader = null;
+		TestingSinkFunction sink = null;
+
+		try {
+			StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+			env.setParallelism(1);
+
+			monitoringFunction = new FileSplitMonitoringFunction<String>(format, hdfsURI,
+				config, FileSplitMonitoringFunction.WatchType.REPROCESS_WITH_APPENDED,
+				env.getParallelism(), INTERVAL);
+
+			TypeInformation<String> typeInfo = TypeExtractor.getInputFormatTypes(format);
+			reader = new FileSplitReadOperator<String>(format, typeInfo, config);
+			sink = new TestingSinkFunction(monitoringFunction);
+
+			DataStream<FileInputSplit> splits = env.addSource(monitoringFunction);
+			splits.transform("FileSplitReader", typeInfo, reader).addSink(sink).setParallelism(1);
+			env.execute();
+
+		} catch (Exception e) {
+			if (!(e.getCause() instanceof SuccessException)) {
+				e.printStackTrace();
+				fail();
+			}
+		}
+	}
+
+	private static int elementCounter = 0;
+	private static Map<Integer, Integer> elementCounters = new HashMap<>();
+	private static Map<Integer, String> collectedContent = new HashMap<>();
+
+	private static class TestingSinkFunction extends RichSinkFunction<String> {
+
+		private final FileSplitMonitoringFunction src;
+
+		TestingSinkFunction(FileSplitMonitoringFunction monitoringFunction) {
+			this.src = monitoringFunction;
+		}
+
+
+		@Override
+		public void open(Configuration parameters) throws Exception {
+			// this sink can only work with DOP 1
+			assertEquals(1, getRuntimeContext().getNumberOfParallelSubtasks());
+		}
+
+		@Override
+		public void close() {
+			src.cancel();
+			try {
+				src.close();
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+
+		@Override
+		public void invoke(String value) throws Exception {
+			synchronized (lock) {
+				int fileIdx = Character.getNumericValue(value.charAt(0));
+
+				Integer counter = elementCounters.get(fileIdx);
+				if(counter == null) {
```

Comment:
yet again a missing space after if. could you please make a *comprehensive* sweep for this?